src/elements/content-explorer/ContentExplorer.js

```diff
@@ -725,6 +729,38 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    /**
+     * Returns an object with newCollection and newSelected properties.
+     * newCollection will be the given collection, with items.selected properties
+     * updated according to the given selected param. newSelected will be the selected
+     * item if it is in newCollection, otherwise undefined.
+     *
+     * @private
+     * @param {Collection} collection - collection that needs to be updated
+     * @param {?BoxItem} selected - item that should be selected in that collection (if present)
+     * @return {Object}
+     */
+    updateCollection(collection: Collection, selected: ?BoxItem): Object {
```

Comment:
Would it make sense to simply call `setState` here since that seems to be how it's used?

Reply:
My thinking on not calling setState in updateCollection is the following scenario: someone wants to set the state of collection, selected, and some other piece(s) of state, and then have a callback function.  This happens in the select function of ContentExplorer. If setState is called in updateCollection, then that callback wouldn't function the same, since collection and selected were set earlier .  Am I thinking about this correctly, or am I misunderstanding how callbacks from setState work?